src/mixed.js

```diff
@@ -382,11 +381,19 @@ const proto = (SchemaType.prototype = {
   },
 
   when(keys, options) {
+    if (arguments.length === 1) {
+      options = keys;
+      keys = '.';
+    } else if (isSchema(keys) || typeof keys === 'function') {
```

Comment:
this is: `when(string(), { then: ... })`? Sorry, I find this to be a confusing api. I realize that Joi has this but it feels like an API wart, i don't want to import. I'm also probably still a 👎 on schema as `is` conditions, but i'll cover that in another reply

Reply:
So you prefer `when({ is: check, then: branch })` over `when(check, { then: branch })`? 
What about `when(check, branch)` (will take one line to add)?

Reply:
i say go with just `when(options)` and `when(keys, options)` its not a lot of code but a far bit of api complexity for minimal gain

Reply:
which is to say i very intentionally trimmed down these overloads when i added `when` originally

Reply:
Ok, removed those.